tests/test_collect_columns.py

```diff
@@ -51,11 +52,11 @@ def test_collect_columns_htseq_with_names():
               datadir / Path("htseq") / Path("sample2.fragments_per_gene")]
     expected_result = pd.DataFrame(data={
         "feature": ["MSTRG.1", "MSTRG.2", "MSTRG.3", "MSTRG.4", "MSTRG.5",
-                    "MSTRG.6", "__no_feature", "__ambiguous",
```

Comment:
Dictionaries do not retain order. This makes testing dictionaries difficult.
However you can extract sets. A set can be compared, and order is not important (unlike a list). 

In this case, having the panda dataframe structure makes testing more difficult. Also I am convinced that we can achieve huge speedups by dropping panda as a dependency alltoghether.